### PR TITLE
Improve CLI

### PR DESCRIPTION
**Pull request by mbatoul**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,80 +1,161 @@
...
+```
 
-- [Intel® oneAPI Data Analytics Library](https://github.com/oneapi-src/oneDAL)
-- [lightgbm](https://lightgbm.readthedocs.io/en/latest/index.html) (gradient boosting library)
-- [XGBoost](https://xgboost.readthedocs.io/en/latest/) (gradient boosting library)
-- [Catboost](https://catboost.ai/) (gradient boosting library)
-- [ONNX Runtime](https://github.com/microsoft/onnxruntime) (inferencing and training accelerator)
+**Run benchmarks for all estimators with profiling**
+```sh
+sklbench run --profile
```
Comment (ogrisel) [inline README.md]: I am not sure it's worth adding a script stub in the PATH when installing the tool. I think using the `-m` flag of `python` is enough:

```
python -m sklearn_benchmarks run --profile
```

File: README.md [hunk 2]
```diff
@@ -1,80 +1,161 @@
...
+In order to setup the environment, you need to have `conda` installed. See instructions [here](https://conda.io/projects/conda/en/latest/user-guide/install/index.html).
 
-These notebooks automatically deployed to `github-pages`. See current results [here](https://mbatoul.github.io/sklearn_benchmarks/).
+To get a local copy up and running follow these simple example steps:
 
-sklearn_benchmarks is used through a command line as described below.
+```sh
+$ git clone -b master --single-branch https://github.com/mbatoul/sklearn_benchmarks
+$ cd sklearn_benchmarks
+$ conda env create --file dev_environment.yml
```
Comment (ogrisel) [inline README.md]: Instead of calling it `dev_environment.yml` it would make more sense to call it `bench_environment.yml`. Otherwise it's confusing the readers' own dev environment.

Some CI scripts might need to be adjusted accordingly

File: README.md [hunk 3]
```diff
@@ -1,80 +1,161 @@
...
+**Run HPO benchmarks only**
+```sh
+sklbench run --strategy hpo
+```
 
-## Getting Started
+**Set a custom results directory**
+```sh
+sklbench run --results-dir ~/dev/sklbench_results
+```
```
Comment (ogrisel) [inline README.md:117]: Could you explain here which results dir is used if this is not passed explicitly? That is a timestamp will be used as result dir.

Comment (ogrisel) [reply README.md:117]: BTW, if the user does not pass a valid timestamp, will this break the reporting tool when calling it for all the sub-folders of a given parent?

If so the reporting tool should probably be made robust to failures of timestamp parsing and just use the raw folder name as section header in the report.

File: config.yml [hunk 4]
```diff
@@ -625,92 +625,101 @@ benchmarking:
...
+          base_lib_name: sklearn_KNeighborsClassifier_brute_force
+          other_lib_name: onnx_sklearn_KNeighborsClassifier_brute_force
+          split_bars_by_parameters:
+            - n_jobs
+
+        # - title: KNeighborsClassifier (KD tree)
+        #   base_lib_name: sklearn_KNeighborsClassifier_kd_tree
+        #   other_lib_name: onnx_sklearn_KNeighborsClassifier_kd_tree
+        #   split_bars_by_parameters:
+        #     - n_jobs
```
Comment (ogrisel) [inline config.yml:686]: Could you please add an inline comment in the config file to explain why this case is commented out?

**Review by ogrisel — COMMENTED** [review]
Some feedback below:

File: slurm_config.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+logs_dir: slurm_logs
+parameters:
+  timeout_min: 70
+  slurm_partition: normal
+  slurm_gpus_per_task: 0
+  slurm_additional_parameters:
+    hint: nomultithread
+  slurm_exclusive: True
+  slurm_cpus_per_task: 40
```
Comment (ogrisel) [inline slurm_config.yml]: If we use `nomultithread` (to disable hyper-threading / SMT), we should target 20 cpus per task to be able to use the base nodes on the margaret cluster.

```suggestion
  slurm_cpus_per_task: 20
```

**Comment by ogrisel** [comment]
I followed the README and could succcessfully create a working bench env, running some benchmarks on a specific estimator with and without slurm and build the HTML report without any problem.

File: config.yml [hunk 6]
```diff
@@ -625,92 +625,89 @@ benchmarking:
...
+      other_library: onnx
+      parameters:
+        n_cols: 2
+      estimators:
+        - title: KNeighborsClassifier (brute force)
+          name: KNeighborsClassifier_brute_force
+          split_bars_by_parameters:
+            - n_jobs
+
+        # - title: KNeighborsClassifier (KD tree)
```
Comment (ogrisel) [inline config.yml:683]:
```suggestion
        # Currently disabled because of a problem with matching parameters
        # when benchmarking against its ONNX version.
        # - title: KNeighborsClassifier (KD tree)
```

**Review by ogrisel — APPROVED** [review]
